gdal/frmts/hdf4/hdf4multidim.cpp

```diff
@@ -1437,6 +1437,12 @@ union ReadFunc
     intn (*pReadData)(int32, int32 [], int32 [], int32 [], VOIDP);
 };
 
+CPL_NOSANITIZE_UNSIGNED_INT_OVERFLOW
+static inline void IncrPointer(GByte*& ptr, GPtrDiff_t nInc, size_t nIncSize)
```

Comment:
Seems weird that allowing an overflow here is OK.  Maybe a comment explaining?  Or at the call site.

Reply:
The overflow is a bit technical here and not a real one. This comes from the fact that GPtrDiff_t is signed, but size_t is not. So from C/C++ standards, GPtrDiff_t is converted to a unsigned integer, and complement-to-2 arithmetic does things right